daemon/userconsent.go

```diff
@@ -0,0 +1,173 @@
+		// nothing to do
+		return
+	}
+
+	consentMode := acc.consentModeFromUserLocation()
+
+	// logout user if in GDPR consent mode
+	if consentMode == consentModeGDPR && acc.authChecker.IsLoggedIn() {
+		if err := acc.doLightLogout(); err != nil {
+			log.Println(internal.WarningPrefix, "failed to perform light logout:", err)
+		}
+		return
+	}
+
+	// standard mode has analytics enabled by default and no required
+	// consent flow, so update the config with `AnalyticsConsent := true`
+	if consentMode == consentModeStandard {
+		if err := acc.setConsentTrue(); err != nil {
+			log.Println(internal.WarningPrefix, "failed to save analytics consent", err)
+		}
+	}
+}
+
+func (acc *AnalyticsConsentChecker) setConsentTrue() error {
+	return acc.cm.SaveWith(func(c config.Config) config.Config {
+		enabled := true
+		c.AnalyticsConsent = &enabled
+		return c
+	})
+}
+
+// IsConsentFlowCompleted reads configuration file and
+// checks if `AnalyticsConsent` field is set.
+func IsConsentFlowCompleted(cm config.Manager) bool {
+	var cfg config.Config
+	if err := cm.Load(&cfg); err != nil {
+		log.Println(internal.ErrorPrefix, "failed to load config when checking consent flow", err)
+		return false
+	}
+	return cfg.AnalyticsConsent != nil
+}
+
+// consentModeFromUserLocation in a happy path, uses Insights API to get user's
+// location and compares it to list of countries in standard mode, if not on the
+// list, then user is in GDPR country.
+//
+// Additionally:
+// - in case of issue with reading configuration, fallback to GDPR mode
+// - if user has KillSwitch enabled, no traffic is going out, fallback to GDPR mode
+// - if there is an issue with making API request, fallback to GDPR mode
+func (acc *AnalyticsConsentChecker) consentModeFromUserLocation() consentMode {
+	var cfg config.Config
+	if err := acc.cm.Load(&cfg); err != nil {
+		log.Println(internal.WarningPrefix, "failed to load config, falling back to GDPR mode:", err)
+		// fallback to strict mode in case of an issue with config
+		return consentModeGDPR
+	}
+
+	// can't determine user location with KS on, fallback to strict mode
+	if cfg.KillSwitch {
+		log.Println(internal.WarningPrefix, "KillSwitch active, falling back to GDPR mode")
+		return consentModeGDPR
+	}
+
+	// fallback to strict mode in case of an issue with API
+	insights, err := acc.insightsAPI.Insights()
+	if err != nil {
+		log.Println(internal.WarningPrefix, "insights api error, falling back to GDRP mode:", err)
+		return consentModeGDPR
+	}
+
+	// fallback to strict mode in case of nil response
+	if insights == nil {
+		log.Println(internal.WarningPrefix, "insigts data is nil, falling back to GDPR mode")
+		return consentModeGDPR
+	}
+
+	mode := modeForCountryCode(countryCode(strings.ToLower(insights.CountryCode)))
+	log.Printf(internal.DebugPrefix+" consent mode for country code '%s': %s", insights.CountryCode, mode)
+	return mode
+}
+
+func (acc *AnalyticsConsentChecker) doLightLogout() error {
```

Comment:
Some documentation on this would be nice.

Reply:
Do you mean doc comments on the method or implementation documentation? 
I updated the specs we have with information how light logout is performed and also created a ticket to write implementation docs with low level details